### ISSUE TITLE
why i can not train ? lt always kill ?

### ISSUE THREAD

**Issue by tanghui2017108** [chenyuntc/simple-faster-rcnn-pytorch#119]
why i can not train ? lt always kill ?
who can help me ?

![2019-04-16 11-16-33屏幕截图](https://user-images.githubusercontent.com/32611793/56179864-ce2d7a00-6039-11e9-88d4-cd783a1e1028.png)

**Comment by charles-wangzi** [comment]
I have the same question. can you resolve it?

**Comment by yanlongbinluck** [comment]
I have the same problem. anyone solve it?

**Comment by zh99li** [comment]
请问你解决这个问题了吗？

**Comment by zh99li** [comment]
> I have the same question. can you resolve it?

请问你解决这个问题了吗？

**Comment by yanlongbinluck** [comment]
> > I have the same question. can you resolve it?
> 
> 请问你解决这个问题了吗？

没有

**Comment by zh99li** [comment]
把train.py中的num_workers设置为0就能解决

**Comment by yanlongbinluck** [comment]
> 把train.py中的num_workers设置为0就能解决

谢谢

**Comment by riciche** [comment]
> 把train.py中的num_workers设置为0就能解决

请问我按您说的做了，但是还是会卡在那里，怎么解决

**Comment by riciche** [comment]
> > 把train.py中的num_workers设置为0就能解决
> 
> 请问我按您说的做了，但是还是会卡在那里，怎么解决

def train(**kwargs):
    opt._parse(kwargs)

    dataset = Dataset(opt)
    print('load data')
    dataloader = data_.DataLoader(dataset, \
                                  batch_size=1, \
                                  shuffle=True, \
                                  # pin_memory=True,
                                  #num_workers=opt.num_workers
                                  num_workers=0)
    testset = TestDataset(opt)
    test_dataloader = data_.DataLoader(testset,
                                       batch_size=1,
                                       #num_workers=opt.test_num_workers,
                                       num_workers=0,
                                       shuffle=False, \
                                       pin_memory=True
                                       )
是这样吗

**Comment by LeonardNiu** [comment]
> why i can not train ? lt always kill ?
> who can help me ?
> 
> ![2019-04-16 11-16-33屏幕截图](https://user-images.githubusercontent.com/32611793/56179864-ce2d7a00-6039-11e9-88d4-cd783a1e1028.png)

which GPU do you use to train?

**Comment by LeonardNiu** [comment]
> why i can not train ? lt always kill ?
> who can help me ?
> 
> ![2019-04-16 11-16-33屏幕截图](https://user-images.githubusercontent.com/32611793/56179864-ce2d7a00-6039-11e9-88d4-cd783a1e1028.png)

I got a runtimeerror, Dataloader worker is killed by signal

**Comment by WangZhaoxue** [comment]
I got the same issue. How can I solve it?

**Comment by Stephenfang51** [comment]
training for around 3-4 epochs and killed
already set test_num for 1000 and num_works = 0, still been killed :(

any good suggestion ?


GPU 1080TI
RAM 32G

**Comment by evilidol** [comment]
I got the same issue, training one epoch killed every time, 
can't find the reason?

anyone solve the problem?

**Comment by Stephenfang51** [comment]
> I got the same issue, training one epoch killed every time,
> can't find the reason?
> 
> anyone solve the problem?

我解决了， 但是我是因为服务器的关系， 本来是共享资源 所以GPU显存不足，后来独占之后就可以了， 这类型的问题只有可能是资源问题， GPU显存或者你内存不够， 很直观

**Comment by TWDH** [comment]
遇见了同样的问题，哪个大大解决了能告诉一下原因或者解决方案嘛？ 应该不是显卡性能和内存的问题

**Comment by evilidol** [comment]
> 遇见了同样的问题，哪个大大解决了能告诉一下原因或者解决方案嘛？ 应该不是显卡性能和内存的问题

应该是显卡和内存的问题，我把pin_memory属性FALSE，就没有出现了

**Comment by Stephenfang51** [comment]
> 遇见了同样的问题，哪个大大解决了能告诉一下原因或者解决方案嘛？ 应该不是显卡性能和内存的问题

建议你把配置都写一下吧 这个估计大神没维护了

**Comment by TWDH** [comment]
@evilidol @Stephenfang51 
非常感谢两位大神，pin_memory改成False之后完美运行不再卡死啦
总结一下遇到训练前几个epoch时卡死的问题：
1.根目录 ---> utils ---> config.py 将文件中相应变量改为 num_workers = 0，test_num_workers = 0
2.根目录 ---> train.py 在train()函数中的test_dataloader(pin_memory=False), 或直接删除pin_memory参数即可
应该是电脑的GPU显存或者内存不能同时处理多线程等问题引起的错误，更改参数减小电脑压力即可

For people who doesn't speak chinese:
Thanks so much for helping~
problem: when run the train.py , system kills the terminal or just crash, try following:
1. root ---> utils ---> config.py change the corresponding varible to num_workers = 0，test_num_workers = 0 
2. root---> train.py, look at the function train(),  change to test_dataloader(pin_memory=False),or just comment out pin_memory and you will be fine.

**Comment by yangninghua** [comment]
mark.thank you

**Comment by yangninghua** [comment]
```
======user config========
{'caffe_pretrain': False,
 'caffe_pretrain_path': 'checkpoints/vgg16_caffe.pth',
 'data': 'voc',
 'debug_file': './debugf',
 'env': 'fasterrcnn-caffe',
 'epoch': 14,
 'load_path': None,
 'lr': 0.001,
 'lr_decay': 0.1,
 'max_size': 1000,
 'min_size': 600,
 'num_workers': 0,
 'plot_every': 100,
 'port': 8097,
 'pretrained_model': 'vgg16',
 'roi_sigma': 1.0,
 'rpn_sigma': 3.0,
 'test_num': 10000,
 'test_num_workers': 0,
 'use_adam': False,
 'use_chainer': False,
 'use_drop': False,
 'voc_data_dir': '/home/spple/data/VOCdevkit/VOC2012/',
 'weight_decay': 0.0005}
==========end============
load data
model construct completed
Setting up a new session...
Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.
99it [00:48,  2.05it/s]> /home/spple/paddle/DeepGlint/deepglint-adv/simple-faster-rcnn-pytorch/train.py(88)train()
     87                 # plot loss
---> 88                 trainer.vis.plot_many(trainer.get_meter_data())
     89 

ipdb> 
```

**Comment by salei0926** [comment]
按照大家说的改了。还是cup占用很高，2000%，应该怎么办啊

**Comment by helenabdr** [comment]
I have the same problem and did what was proposed, but it still doesn't work. Any suggestion?
